packages/test.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+set -e
+
+# Change to packages directory.
+cd "$(dirname "$0")"
+
+TEST_PYTHON=${TEST_PYTHON:-"python"}
+TEST_ENV_DIR=${TEST_ENV_DIR:-`mktemp -d -t gxpkgtestenvXXXXXX`}
+
+virtualenv -p "$TEST_PYTHON" "$TEST_ENV_DIR"
+. "${TEST_ENV_DIR}/bin/activate"
+pip install pytest
+
+# ensure ordered by dependency dag
+PACKAGE_DIRS=(
+    util
+    objectstore
+    job_metrics
+    containers
+    tool_util
+    data
+)
+# containers has no tests, tool_util not yet working 100%,
+# data has many problems quota, tool shed install database, etc..
+RUN_TESTS=(1 1 1 0 0 0 0)
+
+for ((i=0; i<${#PACKAGE_DIRS[@]}; i++)); do
+    package_dir=${PACKAGE_DIRS[$i]}
+    run_tests=${RUN_TESTS[$i]}
+
+    cd "$package_dir"
+    pip install -e .
```

Comment:
Should we add `--index-url https://wheels.galaxyproject.org/simple --extra-index-url https://pypi.python.org/simple` to get bx-python wheels ?
The galaxy-daya install seems to fail with

```
Building wheels for collected packages: bx-python, SQLAlchemy, tzlocal, bagit, python-openid, Tempita
  Building wheel for bx-python (setup.py) ... - error
  ERROR: Complete output from command /tmp/gxpkgtestenvODtzjt/bin/python -u -c 'import setuptools, tokenize;__file__='"'"'/tmp/pip-install-yIvHjU/bx-python/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-UDdsfA --python-tag cp27:
  ERROR: Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-yIvHjU/bx-python/setup.py", line 223, in <module>
      main()
    File "/tmp/pip-install-yIvHjU/bx-python/setup.py", line 74, in main
      raise Exception( "numpy must be installed to build" )
  Exception: numpy must be installed to build
  ----------------------------------------
  ERROR: Failed building wheel for bx-python
```

Reply:
I'd really like these libraries to be as vanilla as possible and avoid this Galaxy-ism. Maybe we could publish our bx-python wheels to PyPI instead? I'll see who I can ping on that (https://github.com/bxlab/bx-python/issues/42).

Reply:
Clearly the best option.